skactiveml/pool/_core_set.py

```diff
@@ -0,0 +1,242 @@
+                X_with_cand = np.concatenate((X_cand, X[selected_samples]), axis=0)
+                n_new_cand = X_cand.shape[0]
+                y_cand = np.full(shape=n_new_cand, fill_value=self.missing_label)
+                y_with_cand = np.concatenate((y_cand, y[selected_samples]), axis=None)
+                mapping = np.arange(n_new_cand)
+                query_indices, utilities = k_greedy_center(X_with_cand, y_with_cand, batch_size, self.random_state_,
+                                                           self.missing_label, mapping, n_new_cand)
+
+        if return_utilities:
+            return query_indices, utilities
+        else:
+            return query_indices
+
+def k_greedy_center(X, y, batch_size, random_state, missing_label=np.nan, mapping=None, n_new_cand=None):
+    """
+     An active learning method that greedily forms a batch to minimize
+     the maximum distance to a cluster center among all unlabeled
+     datapoints.
+     This method is a static method.
+
+     Parameters:
+     ----------
+     X: array-like of shape (n_samples, n_features)
+        Training data set, usually complete, i.e. including the labeled and
+        unlabeled samples
+     selected_samples: np.ndarray of shape (n_selected_samples, )
+        index of datapoints already selects
+     batch_size: int, optional(default=1)
+        The number of samples to be selected in one AL cycle.
+     random_state: np.random.RandomState
+        The random state to use
+     missing_label: scalar or string or np.nan or None, default=np.nan
+        Value to represent a missing label
+     mapping: np.ndarray of shape (n_candidates, ) default None
+        Index array that maps `candidates` to `X`.
+        (`candidates = X[mapping]`)
+     n_new_cand: int or None, default None
+        The number of new candidates that are additionally added to X.
+        Only used for the case, that in the query function with the
+        shape of candidates is (n_candidates, n_feature)
+        
+     Return:
+     ----------
+     query_indices: numpy.ndarry of shape (batch_size, )
+         The query_indices indicate for which candidate sample a label is
+         to queried from the candidates.
+         If candidates in None or of shape (n_candidates), the indexing
+         refers to samples in X.
+         If candidates is of shape (n_candidates, n_features), the indexing
+         refers to samples in candidates.
+     utilities: numpy.ndarray of shape (batch_size, n_samples) or
+         numpy.ndarry of shape (batch_size, n_new_cand)
+         The distance between each data point and its nearest center that used
+         for selecting the next sample.
+         If candidates is None or of shape (n_candidates), the indexing
+         refers to samples in X.
+         If candidates is of shape (n_candidates, n_features), the indexing
+         refers to samples in candidates.
+        """
+    # read the labeled aka selected samples from the y vector
```

Comment:
Could you check whether the parameters are valid? For example:
```
X = check_array(X, allow_nd=True)
y = check_array(y, ensure_2d=False, force_all_finite="allow-nan", dtype=None)
check_consistent_length(X, y)    
```
Cf. `base.by` to see which checks are possible.